### ISSUE TITLE
Invalid shape for monophonic audio (augmentation) + Dataset 4 question

### ISSUE THREAD

**Issue by KatieBelli** [ZFTurbo/Music-Source-Separation-Training#135]
When I want to use augmentation this error appears (choosing only the channel_shuffle works fine though):

![Image](https://github.com/user-attachments/assets/6359bf99-c31f-42a6-9f09-d4fa727fd1ef)

I have mono files in my training dataset which were converted into stereo files before training (left and right channel containing the same audio).

What do I need to do to get rid of those 2 errors?

I use Prodigy as optimizer.

Here are the augmentations I intend to use (might want to use the other augmentations for other trainings):

```
augmentations:
  enable: true # enable or disable all augmentations (to fast disable if needed)
  loudness: true # randomly change loudness of each stem on the range (loudness_min; loudness_max)
  loudness_min: 0.5
  loudness_max: 1.5
  
  all:
    channel_shuffle: 0.5 # Set 0 or lower to disable

    # pedalboard pitch shift block
    pedalboard_pitch_shift: 0.01
    pedalboard_pitch_shift_semitones_min: -8
    pedalboard_pitch_shift_semitones_max: 5

    # pedalboard resample block
    pedalboard_resample: 0.01
    pedalboard_resample_target_sample_rate_min: 4000
    pedalboard_resample_target_sample_rate_max: 44100

  Target:
    pitch_shift: 0.1
    pitch_shift_min_semitones: -8
    pitch_shift_max_semitones: 5
    seven_band_parametric_eq: 0.25
    seven_band_parametric_eq_min_gain_db: -9
    seven_band_parametric_eq_max_gain_db: 9
    tanh_distortion: 0.1
    tanh_distortion_min: 0.1
    tanh_distortion_max: 0.7
  Other:
    pitch_shift: 0.1
    pitch_shift_min_semitones: -8
    pitch_shift_max_semitones: 5
    seven_band_parametric_eq: 0.25
    seven_band_parametric_eq_min_gain_db: -9
    seven_band_parametric_eq_max_gain_db: 9
    tanh_distortion: 0.1
    tanh_distortion_min: 0.1
    tanh_distortion_max: 0.7
```

**Other question:** Does this cause issues with the sync of Dataset 4? 

`min_mean_abs: 0.000`

My audio files are synced and them staying synced while training is crucial.
(The target stem and the other stem sometimes have silence but it's not many seconds. I had to change it to 0.000 or else it always shows an error during training.)

**Comment by iver56** [comment]
Check your librosa version. It might be too old. Upgrading it would probably fix your problem. Or, alternatively, you could try downgrading numpy to 1.23 or 1.24.

**Comment by iver56** [comment]
Hey by the way, you should upgrade audiomentations too. You don't have the latest version.

**Comment by KatieBelli** [comment]
Thank you, I upgraded librosa and audiomentations audio to the latest version but the error remains unchanged.

**Comment by iver56** [comment]
If you could share your list of dependency versions, e.g. the output of `pip freeze`, it would be easier to help

**Comment by KatieBelli** [comment]
> If you could share your list of dependency versions, e.g. the output of `pip freeze`, it would be easier to help

Sure, here they are:

absl-py==1.4.0
accelerate==1.3.0
aiofiles==23.1.0
aiohttp==3.8.4
aiosignal==1.3.1
alembic==1.13.2
altair==4.2.2
annotated-types==0.7.0
ansicon==1.89.0
antlr4-python3-runtime==4.9.3
anyio==3.6.2
appdirs==1.4.4
arrow==1.2.3
asteroid==0.7.0
asteroid-filterbanks==0.4.0
astunparse==1.6.3
async-timeout==4.0.2
attrs==23.1.0
audiomentations==0.39.0
audioread==2.1.9
auraloss==0.4.0
beartype==0.14.1
beautifulsoup4==4.12.2
bitarray==2.7.6
bitsandbytes==0.43.0
bitsandbytes-windows==0.37.5
blessed==1.20.0
blinker==1.6.2
cached-property==2.0.1
cachetools==5.3.0
certifi==2022.6.15
cffi==1.15.1
charset-normalizer==2.1.0
click==8.1.3
cloudpickle==2.2.0
cm-time==0.1.2
colorama==0.4.5
coloredlogs==15.0.1
colorlog==6.8.2
contextlib2==21.6.0
contourpy==1.0.7
croniter==1.3.14
cycler==0.11.0
Cython==0.29.32
dateutils==0.6.12
decorator==5.1.1
deepdiff==6.3.0
defusedxml==0.7.1
demucs==4.0.0
diffq==0.2.3
docker-pycreds==0.4.0
dora-search==0.1.11
efficientnet-pytorch==0.7.1
einops==0.6.1
entrypoints==0.4
eval_type_backport==0.2.2
exceptiongroup==1.2.2
fairseq==0.12.2
faiss-cpu==1.7.3
fastapi==0.98.0
ffmpeg==1.4
ffmpeg-downloader==0.2.0
ffmpeg-python==0.2.0
ffmpy==0.3.0
filelock==3.9.0
Flask==2.3.1
Flask-Cors==3.0.10
flatbuffers==23.5.26
fonttools==4.38.0
frozenlist==1.3.3
fsspec==2024.2.0
future==0.18.2
gast==0.4.0
gitdb==4.0.11
GitPython==3.1.43
google-auth==2.17.3
google-auth-oauthlib==1.0.0
google-pasta==0.2.0
gradio==3.14.0
gradio_client==0.1.3
greenlet==3.0.3
grpcio==1.54.0
h11==0.12.0
h5py==3.9.0
httpcore==0.15.0
httpx==0.23.0
huggingface-hub==0.27.1
humanfriendly==10.0
hydra-core==1.0.7
idna==3.3
imageio==2.28.0
importlib-metadata==6.6.0
importlib-resources==5.12.0
inflect==5.3.0
iniconfig==2.0.0
inquirer==3.1.3
iso8601==1.0.2
itsdangerous==2.1.2
Jinja2==3.1.2
jinxed==1.2.0
joblib==1.1.0
json5==0.9.14
jsonpatch==1.32
jsonpointer==2.3
jsonschema==4.17.3
julius==0.2.7
keras==2.13.1
keyboard==0.13.5
kiwisolver==1.4.4
lameenc==1.4.1
lazy_loader==0.2
libaio==0.9.2
libclang==16.0.6
librosa==0.10.2.post1
lightning==2.0.2
lightning-cloud==0.5.34
lightning-utilities==0.8.0
linkify-it-py==2.0.0
llvmlite==0.39.0
loralib==0.1.2
lxml==4.9.2
m3u8==3.0.0
Mako==1.3.5
Markdown==3.4.3
markdown-it-py==2.2.0
MarkupSafe==2.1.2
matplotlib==3.8.2
matplotlib-inline==0.1.6
mdit-py-plugins==0.3.3
mdurl==0.1.2
mir-eval==0.7
ml-collections==0.1.1
mpmath==1.2.1
msgpack==1.0.7
multidict==6.0.4
munch==4.0.0
mutagen==1.45.1
networkx==3.0
numba==0.56.4
numpy==1.23.5
numpy-minmax==0.3.1
numpy-rms==0.4.2
nvidia-cublas-cu11==11.11.3.6
nvidia-cuda-nvrtc-cu11==11.8.89
nvidia-cudnn-cu11==8.9.5.29
oauthlib==3.2.2
omegaconf==2.2.3
onnx==1.13.1
onnx2torch==1.5.11
onnxoptimizer==0.3.13
onnxruntime-gpu==1.15.1
onnxsim==0.4.24
opencv-python==4.8.1.78
openunmix==1.2.1
opt-einsum==3.3.0
optuna==3.6.1
ordered-set==4.1.0
orjson==3.8.10
packaging==21.3
pandas==2.0.1
pb-bss-eval==0.0.2
Pebble==5.0.3
pedalboard==0.8.9
pesq==0.0.4
Pillow==9.5.0
platformdirs==4.3.6
pluggy==1.5.0
pooch==1.6.0
portalocker==2.7.0
praat-parselmouth==0.4.3
pretrainedmodels==0.7.4
primePy==1.3
prodigyopt==1.1.2
progressbar==2.5
protobuf==3.20.3
psutil==5.9.5
pyarrow==16.0.0
pyasn1==0.5.0
pyasn1-modules==0.3.0
PyAudio==0.2.14
pycparser==2.21
pycryptodome==3.18.0
pycryptodomex==3.15.0
pydantic==2.10.5
pydantic_core==2.27.2
pydeck==0.9.0
pydub==0.25.1
Pygments==2.15.1
PyInputPlus==0.2.12
PyJWT==2.6.0
pynndescent==0.5.8
pyparsing==3.0.9
PyQt5==5.15.10
PyQt5-Qt5==5.15.2
PyQt5-sip==12.13.0
pyreadline3==3.4.1
pyrsistent==0.19.3
PySimpleGUI==4.60.4
PySimpleValidate==0.2.12
pystoi==0.4.1
pytest==8.3.4
pytest-mock==3.14.0
python-dateutil==2.8.2
python-editor==1.0.4
python-multipart==0.0.6
python-stretch==0.2.0
pytorch-lightning==2.0.2
pytorch-ranger==0.1.1
pytz==2023.3
PyWavelets==1.4.1
pywin32==306
pyworld==0.3.3
PyYAML==6.0
pyzmq==25.0.0
readchar==4.0.5
regex==2023.3.23
requests==2.28.1
requests-oauthlib==1.3.1
resampy==0.4.2
retrying==1.3.3
rfc3986==1.5.0
rich==13.3.4
rotary-embedding-torch==0.3.5
rsa==4.9
sacrebleu==2.3.1
safetensors==0.5.2
scikit-image==0.20.0
scikit-learn==1.0.2
scikit-maad==1.3.12
scipy==1.12.0
segmentation-models-pytorch==0.3.3
semantic-version==2.10.0
sentry-sdk==2.20.0
setproctitle==1.3.4
six==1.16.0
smmap==5.0.1
sniffio==1.3.0
so-vits-svc-fork @ git+https://github.com/voicepaw/so-vits-svc-fork.git@16fa0374ae9b606509bcd3892a6f2e56a23567a8
sounddevice==0.4.3
soundfile==0.12.1
soupsieve==2.4.1
soxr==0.3.7
spafe==0.3.2
SQLAlchemy==2.0.32
starlette==0.27.0
starsessions==1.3.0
stdiomask==0.0.6
streamlit==1.34.0
submitit==1.4.5
sympy==1.11.1
tabulate==0.9.0
tenacity==8.3.0
tensorboard==2.13.0
tensorboard-data-server==0.7.0
tensorboard-plugin-wit==1.8.1
tensorboardX==2.6
tensorflow==2.13.0
tensorflow-estimator==2.13.0
tensorflow-intel==2.13.0
tensorflow-io-gcs-filesystem==0.31.0
termcolor==2.3.0
threadpoolctl==3.1.0
tifffile==2023.4.12
timm==0.9.2
tkinterdnd2==0.3.0
tokenizers==0.15.2
toml==0.10.2
tomli==2.2.1
toolz==0.12.0
torch==2.1.0+cu121
torch-audiomentations==0.12.0
torch-log-wmse==0.2.9
torch-optimizer==0.1.0
torch-stoi==0.2.3
torch_pitch_shift==1.2.5
torchaudio==2.1.0+cu121
torchcrepe==0.0.20
torchfile==0.1.0
torchlibrosa==0.1.0
torchmetrics==0.11.4
torchseg==0.0.1a1
torchvision==0.16.0
tornado==6.2
tqdm==4.66.1
tqdm-joblib==0.0.3
traitlets==5.9.0
transformers==4.35.2
treetable==0.2.5
typing_extensions==4.12.2
tzdata==2023.3
uc-micro-py==1.0.1
umap-learn==0.5.2
Unidecode==1.3.2
urllib3==1.26.20
uvicorn==0.21.1
vinted-downloader==2.2.0
visdom==0.1.8.9
voicefixer @ git+https://github.com/haoheliu/voicefixer.git@ad4304594ef71ae47303f3cf9a8791584fcd7d67
wandb==0.19.4
watchdog==4.0.0
wcwidth==0.2.6
webrtcvad==2.0.10
websocket-client==1.4.2
websockets==11.0.2
Werkzeug==2.3.0
wrapt==1.15.0
wxPython==4.2.2
yarl==1.9.2
zipp==3.15.0

**Comment by iver56** [comment]
Further, you could copy (text, not screenshot) the error(s) you get when running with this newer set of dependencies. Please double-check that your code runs in the correct environment with the upgraded packages. If you are in fact using the latest audiomentations version, by default it shouldn't use librosa for the pitch shifting.

The two errors you posted in your screenshot above suggested you were using an old librosa version. What I know:
* In librosa<0.9.0 pitch_shift doesn't natively support multichannel audio, and would complain when given stereo audio
* Old librosa versions would use the deprecated `np.float` alias, incompatible with modern numpy versions, ref https://github.com/librosa/librosa/issues/1350

**Comment by KatieBelli** [comment]
> Further, you could copy (text, not screenshot) the error(s) you get when running with this newer set of dependencies. Please double-check that your code runs in the correct environment with the upgraded packages. If you are in fact using the latest audiomentations version, by default it shouldn't use librosa for the pitch shifting.
> 
> The two errors you posted in your screenshot above suggested you were using an old librosa version. What I know:
> 
>     * In librosa<0.9.0 pitch_shift doesn't natively support multichannel audio, and would complain when given stereo audio
> 
>     * Old librosa versions would use the deprecated `np.float` alias, incompatible with modern numpy versions, ref [[DeprecationWarning] np aliases of builtin types deprecated in numpy 1.20 librosa/librosa#1350](https://github.com/librosa/librosa/issues/1350)

I found the solution: I had 2 different Python versions installed. I removed them completely (incl. residues) and installed Python 3.13.2 + reinstalled the requirements.txt. Now the training runs without issues. Thank you very much!

I'd love to use the latest version of audiomentations but somehow it can't be installed. Could this be due to my current Python version?

> C:\Users\User\Music\Machine Learning\Music-Source-Separation-Training-main>pip install -r requirements.txt
> Requirement already satisfied: torch>=2.0.1 in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 1)) (2.6.0+cu126)
> Requirement already satisfied: torchaudio in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 2)) (2.6.0+cu126)
> Requirement already satisfied: numpy in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 3)) (2.1.3)
> Requirement already satisfied: pandas in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 4)) (2.2.3)
> Requirement already satisfied: scipy in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 5)) (1.15.2)
> Requirement already satisfied: soundfile in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 6)) (0.13.1)
> Requirement already satisfied: ml_collections in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 7)) (1.0.0)
> Requirement already satisfied: tqdm in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 8)) (4.67.1)
> Requirement already satisfied: segmentation_models_pytorch==0.3.3 in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 9)) (0.3.3)
> Requirement already satisfied: timm==0.9.2 in c:\users\User\appdata\local\programs\python\python313\lib\site-packages (from -r requirements.txt (line 10)) (0.9.2)
> ERROR: Could not find a version that satisfies the requirement audiomentations==0.39.0 (from versions: 0.1.0, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.12.1, 0.13.0, 0.14.0, 0.15.0, 0.16.0, 0.17.0, 0.18.0, 0.19.0, 0.20.0, 0.21.0, 0.22.0, 0.23.0, 0.24.0, 0.25.0, 0.25.1, 0.26.0, 0.27.0, 0.28.0, 0.29.0, 0.30.0, 0.31.0, 0.32.0, 0.33.0, 0.34.1, 0.35.0, 0.36.0, 0.36.1)
> ERROR: No matching distribution found for audiomentations==0.39.0

**Comment by iver56** [comment]
Audiomentations does not support Python 3.13 at the moment. Although it is sometimes smart to use the newest Python version, it is often wise to stay on a slightly older version for compatibility reasons. Try 3.12
